js-packages/@prestojs/viewmodel/src/ViewModelCache.ts

```diff
@@ -622,19 +708,26 @@ export default class ViewModelCache<T extends ViewModelInterface<any, any>> {
      * @param removeNulls whether to remove entries that have no record in the cache. Defaults to true.
      * @returns an array of the cached records. Any records not found will be in the array as a null value if `removeNulls` is false otherwise they will be removed.
      */
-    getList(pks: PrimaryKey[], fieldNames: string[], removeNulls?: boolean): (T | null)[];
-    getList(
-        pksOrRecords: PrimaryKey[],
-        fieldNames?: string[] | boolean,
+    getList<FieldNames extends keyof ViewModelClassType['fields']>(
+        pks: PrimaryKey[],
+        fieldNames: readonly FieldNames[],
+        removeNulls?: boolean
+    ): (PartialViewModel<ViewModelClassType, FieldNames> | null)[];
+    getList<
+        FieldNames extends keyof ViewModelClassType['fields'],
+        T extends InstanceType<ViewModelClassType>
+    >(
+        pksOrRecords: (T | PrimaryKey)[],
+        fieldNames?: readonly FieldNames[] | boolean,
```

Comment:
fieldNames being true isnt mentioned in the docstring?

Reply:
fieldNames is required to be an array but only if you provide primary keys, not records. this signature here is the combined signature of the overloads so has to accomodate everything and hence when `fieldNames` is `boolean` it's actually the `removeNulls` option. Probably clearer looking at then documentation: https://prestojs.com/docs/viewmodel/ViewModelCache/#method-getList

Reply:
ah i see